src/sidebar/components/test/markdown-editor-test.js

```diff
@@ -0,0 +1,248 @@
+          `ToolbarButton[title="${command}"] > button`
+        );
+        const input = wrapper.find('textarea').getDOMNode();
+        input.selectionStart = 0;
+        input.selectionEnd = 4;
+
+        button.simulate('click');
+
+        assert.calledWith(onEditText, {
+          text: 'formatted text',
+        });
+        const [formatFunction, ...args] = effect;
+        assert.calledWith(
+          formatFunction,
+          sinon.match({ text: 'test', selectionStart: 0, selectionEnd: 4 }),
+          ...args
+        );
+      });
+
+      if (key) {
+        it('applies formatting when shortcut key is pressed', () => {
+          const onEditText = sinon.stub();
+          const wrapper = mount(
+            <MarkdownEditor text="test" onEditText={onEditText} />
+          );
+          const input = wrapper.find('textarea');
+          input.getDOMNode().selectionStart = 0;
+          input.getDOMNode().selectionEnd = 4;
+
+          input.simulate('keydown', {
+            ctrlKey: true,
+            key,
+          });
+
+          assert.calledWith(onEditText, {
+            text: 'formatted text',
+          });
+          const [formatFunction, ...args] = effect;
+          assert.calledWith(
+            formatFunction,
+            sinon.match({ text: 'test', selectionStart: 0, selectionEnd: 4 }),
+            ...args
+          );
+        });
+      }
+    });
+  });
+
+  [
+    {
+      // Shortcut letter but without ctrl key.
+      key: 'b',
+      ctrlKey: false,
+    },
+    {
+      // Ctrl key with non-shortcut letter
+      key: 'w',
+      ctrlKey: true,
+    },
+  ].forEach(({ ctrlKey, key }) => {
+    it('does not apply formatting when a non-shortcut key is pressed', () => {
+      const onEditText = sinon.stub();
+      const wrapper = mount(
+        <MarkdownEditor text="test" onEditText={onEditText} />
+      );
+      const input = wrapper.find('textarea');
+
+      input.simulate('keydown', {
+        ctrlKey,
+        key,
+      });
+
+      assert.notCalled(onEditText);
+    });
+  });
+
+  it('calls `onEditText` callback when text is changed', () => {
+    const onEditText = sinon.stub();
+    const wrapper = mount(
+      <MarkdownEditor text="test" onEditText={onEditText} />
+    );
+    const input = wrapper.find('textarea').getDOMNode();
+    input.value = 'changed';
+    wrapper.find('textarea').simulate('input');
+    assert.calledWith(onEditText, {
+      text: 'changed',
+    });
+  });
+
+  it('enters preview mode when Preview button is clicked', () => {
+    const wrapper = mount(<MarkdownEditor text="test" />);
+
+    const previewButton = wrapper
+      .find('button')
+      .filterWhere(el => el.text() === 'Preview');
+    previewButton.simulate('click');
+
+    assert.isFalse(wrapper.find('textarea').exists());
+    assert.isTrue(wrapper.find('MarkdownView').exists());
+    wrapper
+      .find('button')
+      .filterWhere(el => el.text() !== 'Write')
```

Comment:
Is there a reason that this conditional can't be the (easier-to-make-sense-of) `=== 'Preview'` ?

Reply:
In this case we're trying to get all the toolbar buttons _except_ the one that is labeled "Write"/"Preview" to check that they are disabled in preview mode.